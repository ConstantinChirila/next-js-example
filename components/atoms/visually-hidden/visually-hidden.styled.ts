import styled from 'styled-components'

export const StyledVisuallyHidden = styled.p`
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 0.1rem;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
`
