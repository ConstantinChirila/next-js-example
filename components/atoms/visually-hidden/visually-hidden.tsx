import React, { ReactElement } from 'react'

import { StyledVisuallyHidden } from './visually-hidden.styled'
import { TVisuallyHiddenProperties } from './visually-hidden.types'

export function VisuallyHidden(
  properties: TVisuallyHiddenProperties
): ReactElement {
  return <StyledVisuallyHidden {...properties} />
}
