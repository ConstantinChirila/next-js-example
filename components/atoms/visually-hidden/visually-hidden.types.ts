import { AllHTMLAttributes, ElementType } from 'react'

export type TVisuallyHiddenProperties = AllHTMLAttributes<HTMLDivElement> & {
  as?: ElementType
}
