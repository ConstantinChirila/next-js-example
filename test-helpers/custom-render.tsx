import { PropsWithChildren, ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { QueryClientProvider, QueryClient } from 'react-query'

import { defaultTheme } from '@tooploox-test/theme'
import { reactQueryDefaults } from '@tooploox-test/configuration'

function DefaultThemeProviderWithReactQuery({
  children,
}: PropsWithChildren<{}>): ReactElement {
  const queryClient = new QueryClient({
    ...reactQueryDefaults,
  })

  return (
    <ThemeProvider theme={defaultTheme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  )
}

function DefaultThemeProvider({
  children,
}: PropsWithChildren<{}>): ReactElement {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

function renderWithReactQueryContext(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProviderWithReactQuery, ...options })
}

function renderWithTheme(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProvider, ...options })
}

export { renderWithTheme, renderWithReactQueryContext }
