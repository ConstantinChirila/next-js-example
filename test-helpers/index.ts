export * from './custom-render'
export * from './setup'
