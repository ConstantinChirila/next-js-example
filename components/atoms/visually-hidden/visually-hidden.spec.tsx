import React from 'react'

import { VisuallyHidden } from './visually-hidden'
import { renderWithTheme } from '../../../test-helpers'

describe('VisuallyHidden', () => {
  it('should match a snapshot', () => {
    const { container } = renderWithTheme(
      <VisuallyHidden>hello</VisuallyHidden>
    )
    expect(container).toMatchSnapshot()
  })
})
