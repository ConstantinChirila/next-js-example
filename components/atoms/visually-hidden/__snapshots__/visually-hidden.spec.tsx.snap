// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisuallyHidden should match a snapshot 1`] = `
.c0 {
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

<div>
  <span
    class="c0"
  >
    hello
  </span>
</div>
`;
